TumbleBitSetup.Tests/ProtocolTests.cs

```diff
@@ -399,12 +395,9 @@ public bool _ShortN(BigInteger Exp, int shortKeySize, int longKeySize)
             // A case where "shortKeySize"-bits N is used for proving and "longKeySize"-bits is needed for verifying.
             var keyPair = new RsaKey(Exp, shortKeySize);
 
-            var privKey = keyPair._privKey;
-            var pubKey = new RsaPubKey(keyPair);
-
-            byte[][] signature = PermutationTest.Proving(privKey.P, privKey.Q, privKey.PublicExponent, alpha, ps);
+            var signature = keyPair.CreatePermutationTest(setup);
             // passing "longKeySize" as the keySize.
-            return PermutationTest.Verifying(pubKey, signature, alpha, longKeySize, ps);
+            return keyPair.PublicKey.VerifyPermutationTestProof(signature, setup);
```

Comment:
I am away from my computer today but one thing @osagga to check. The keylength passed to the verifier and used in the first few checks of the verifying algorithm should be the system parameter specified in the setup.  It should not be derived from the public key passed to the verification algorithm.